packages/rum-core/src/domain/assembly.ts

```diff
@@ -162,8 +164,19 @@ function needToAssembleWithAction(
 }
 
 function getSessionType() {
-  return navigator.userAgent.indexOf('DatadogSynthetics') === -1 &&
-    (window as BrowserWindow)._DATADOG_SYNTHETICS_BROWSER === undefined
+  return navigator.userAgent.indexOf('DatadogSynthetics') === -1 && !getSyntheticsContext()
     ? SessionType.USER
     : SessionType.SYNTHETICS
 }
+
+function getSyntheticsContext() {
+  const testId = (window as BrowserWindow)._DATADOG_SYNTHETICS_PUBLIC_ID
+  const resultId = (window as BrowserWindow)._DATADOG_SYNTHETICS_RESULT_ID
```

Comment:
Should we handle the case when some of these variables is not a string? I know it shouldn't normally happen but if it does, what would be the impact? generating the events with wrong format, how would the backend handle it?